demos/python_demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.py

```diff
@@ -15,239 +15,593 @@
-    # Read IR
-    log.info("Loading network")
-    net = ie.read_network(args.model, os.path.splitext(args.model)[0] + ".bin")
-
-    if "CPU" in args.device:
-        supported_layers = ie.query_network(net, "CPU")
-        not_supported_layers = [l for l in net.layers.keys() if l not in supported_layers]
-        if len(not_supported_layers) != 0:
-            log.error("Following layers are not supported by the plugin for specified device {}:\n {}".
-                      format(args.device, ', '.join(not_supported_layers)))
-            log.error("Please try to specify cpu extensions library path in sample's command line parameters using -l "
-                      "or --cpu_extension command line argument")
-            sys.exit(1)
-
-    img_info_input_blob = None
-    feed_dict = {}
-    for blob_name in net.input_info:
-        if len(net.input_info[blob_name].input_data.shape) == 4:
-            input_blob = blob_name
-        elif len(net.input_info[blob_name].input_data.shape) == 2:
-            img_info_input_blob = blob_name
-        else:
-            raise RuntimeError("Unsupported {}D input layer '{}'. Only 2D and 4D input layers are supported"
-                               .format(len(net.input_info[blob_name].input_data.shape), blob_name))
 
-    output_postprocessor = get_output_postprocessor(net)
-
-    log.info("Loading IR to the plugin...")
-    exec_net = ie.load_network(network=net, num_requests=2, device_name=args.device)
-    # Read and pre-process input image
-    n, c, h, w = net.input_info[input_blob].input_data.shape
-    if img_info_input_blob:
-        feed_dict[img_info_input_blob] = [h, w, 1]
-
-    if args.input == 'cam':
-        input_stream = 0
-    else:
-        input_stream = args.input
-    cap = cv2.VideoCapture(input_stream)
-    assert cap.isOpened(), "Can't open " + str(input_stream)
+    config_user_specified, config_min_latency = get_plugin_configs(args.device, args.num_streams, args.num_threads)
 
+    labels_map = None
     if args.labels:
         with open(args.labels, 'r') as f:
             labels_map = [x.strip() for x in f]
-    else:
-        labels_map = None
 
-    cur_request_id = 0
-    next_request_id = 1
+    log.info('Loading network...')
+    completed_request_results = {}
+    modes = cycle(Modes)
+    prev_mode = mode = next(modes)
+    log.info('Using {} mode'.format(mode.name))
+    mode_info = {mode: ModeInfo()}
+    exceptions = []
+
+    exec_nets = {
+        Modes.USER_SPECIFIED:
+            Detector(ie, args.model, device=args.device, plugin_config=config_user_specified,
+                     results=completed_request_results, max_num_requests=args.num_infer_requests,
+                     labels_map=labels_map, keep_aspect_ratio_resize=args.keep_aspect_ratio,
+                     caught_exceptions=exceptions),
+        Modes.MIN_LATENCY:
+            Detector(ie, args.model, device=args.device.split(':')[-1].split(',')[0], plugin_config=config_min_latency,
```

Comment:
Why is the device different in this case?

Reply:
Once again this came from a yolov3 demo. I believe that the idea is to use only primary device for a hetero regime to reduce the latency.